### PR TITLE
Sprint1

### PR DESCRIPTION
**Pull request by adbusnel**
delivery of sprint 1
- dashboards for student, teacher, admin and adm
- login/logout
- router
- forgot password
- landing page

### CODE REVIEW

File: .gitignore
Comment (Exiels) [inline .gitignore:1]: ?

File: LICENSE
Comment (Exiels) [inline LICENSE:1]: ?

File: README.md
Comment (Exiels) [inline README.md:1]: ?

File: package.json [hunk 1]
```diff
@@ -4,14 +4,16 @@
   "private": true,
   "dependencies": {
     "@testing-library/jest-dom": "^5.16.5",
-    "@testing-library/react": "^14.0.0",
-    "@testing-library/user-event": "^14.4.3",
+    "@testing-library/react": "^13.4.0",
+    "@testing-library/user-event": "^13.5.0",
```
Comment (Exiels) [inline package.json:8]: Why downgrade ?

File: src/Components/Sidebar/sidebar.scss [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        display: flex;
+        align-items: center;
+        text-decoration: none;
+        color: #fff;
+        font-size: 20px; /* Ajustement de la taille de police */
+        margin-left: 10px; /* Ajout d'une marge entre l'icône et le texte */
+
+        .sidebar-menu-item-icon {
+          margin-right: 10px;
+          font-size: 24px; /* Ajustement de la taille de l'icône */
```
Comment (Exiels) [inline src/Components/Sidebar/sidebar.scss:56]: Comments in english

File: src/Users/Public/ForgottenPasswordPage.jsx [hunk 3]
```diff
@@ -31,12 +31,13 @@ export default function ForgottenPasswordPage () {
         body: JSON.stringify(payload)
       })
 
-      const data = await response.json()
-
-      if (response.ok) {
-        setMessage(`${data.message}`)
+      // const data = await response.json();
+      console.log(response)
+      // console.log("????");
```
Comment (Exiels) [inline src/Users/Public/ForgottenPasswordPage.jsx:36]: Remove debug log

File: src/Users/Public/loginPage.jsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  const handleLogin = async (event) => {
+    event.preventDefault()
+
+    // Vérifier que l'email est valide avant de faire la requête
+    if (!validateEmail(email)) {
+      setMessage('Email is not valid')
+      return
+    }
+
+    // Vérifier qu'un password a été entré avant de faire la requête
```
Comment (Exiels) [inline src/Users/Public/loginPage.jsx:48]: Comments in english

File: src/Users/Student/dashboard_student.scss
Comment (Exiels) [inline src/Users/Student/dashboard_student.scss:1]: Comments in english and remove unused commented data

File: src/Users/Teacher/dashboard_teacher.scss
Comment (Exiels) [inline src/Users/Teacher/dashboard_teacher.scss:1]: Comments in english and remove unused commented data

File: src/linter-nonfixable
Comment (Exiels) [inline src/linter-nonfixable:1]: Remove this file

File: src/tests/Sidebar.test.jsx
Comment (Exiels) [inline src/tests/Sidebar.test.jsx:1]: Comments in english

File: src/tests/dashboardStudent.test.jsx
Comment (Exiels) [inline src/tests/dashboardStudent.test.jsx:1]: Comments in english

File: src/tests/dashboardTeacher.test.jsx
Comment (Exiels) [inline src/tests/dashboardTeacher.test.jsx:1]: Comments in english

**Review by Exiels — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (adbusnel) [reply package.json:8]: les tests de @Leo-Nord étaient supposés fonctionner avec l'ancienne version, on a des erreurs en plus sinon

**Review by Exiels — APPROVED** [review]

**Review by Exiels — APPROVED** [review]

**Pull request merged**